cmake/rocprim-backward-compat.cmake

```diff
@@ -0,0 +1,309 @@
+set(PROJECT_WRAPPER_WARP_RELPATH        "${include_statements}#include \"../../../../include/rocprim/warp")
+set(PROJECT_WRAPPER_BLK_DET_RELPATH     "${include_statements}#include \"../../../../../include/rocprim/block/detail")
+set(PROJECT_WRAPPER_DEV_DET_RELPATH     "${include_statements}#include \"../../../../../include/rocprim/device/detail")
+set(PROJECT_WRAPPER_DEV_SPE_RELPATH     "${include_statements}#include \"../../../../../include/rocprim/device/specialization")
+set(PROJECT_WRAPPER_ITE_DET_RELPATH     "${include_statements}#include \"../../../../../include/rocprim/iterator/detail")
+set(PROJECT_WRAPPER_WAR_DET_RELPATH     "${include_statements}#include \"../../../../../include/rocprim/warp/detail")
+
+#Include Guard for wrapper (include) files generated
+set(PROJECT_WRAPPER_MAIN_INC_GUARD      "${include_guard}ROCPRIM_WRAPPER_MAIN_INC")
+set(PROJECT_WRAPPER_BLOCK_GUARD     	"${include_guard}ROCPRIM_WRAPPER_BLOCK_INC")
+set(PROJECT_WRAPPER_DETAIL_GUARD        "${include_guard}ROCPRIM_WRAPPER_DETAIL_INC")
+set(PROJECT_WRAPPER_DEVICE_GUARD        "${include_guard}ROCPRIM_WRAPPER_DEVICE_INC")
+set(PROJECT_WRAPPER_INTRINSICS_GUARD    "${include_guard}ROCPRIM_WRAPPER_INTRINSICS_INC")
+set(PROJECT_WRAPPER_ITERATOR_GUARD      "${include_guard}ROCPRIM_WRAPPER_ITERATOR_INC")
+set(PROJECT_WRAPPER_TYPES_GUARD         "${include_guard}ROCPRIM_WRAPPER_TYPES_INC")
+set(PROJECT_WRAPPER_THREAD_GUARD        "${include_guard}ROCPRIM_WRAPPER_THREAD_INC")
+set(PROJECT_WRAPPER_WARP_GUARD          "${include_guard}ROCPRIM_WRAPPER_WARP_INC")
+set(PROJECT_WRAPPER_BLK_DET_GUARD       "${include_guard}ROCPRIM_WRAPPER_BLK_DET_INC")
+set(PROJECT_WRAPPER_DEV_DET_GUARD       "${include_guard}ROCPRIM_WRAPPER_DEV_DET_INC")
+set(PROJECT_WRAPPER_DEV_SPE_GUARD       "${include_guard}ROCPRIM_WRAPPER_DEV_SPE_INC")
+set(PROJECT_WRAPPER_ITE_DET_GUARD       "${include_guard}ROCPRIM_WRAPPER_ITE_DET_INC")
+set(PROJECT_WRAPPER_WAR_DET_GUARD       "${include_guard}ROCPRIM_WRAPPER_WAR_DET_INC")
+
+
+
+# Function for Generating Wrapper Headers for Backward Compatibilty
+# Generates wrapper for all *.h files in include/rocprim folder
+# No Arguments to function.
+# Wrapper are generated under rocprim/include/rocprim.
+function (package_gen_bkwdcomp_hdrs)
+	# Get list of *.h files in folder include/rocprim 
+	file(GLOB include_files ${PROJECT_FREORG_MAIN_INC_DIR}/*.hpp)
+	# Convert the list of files into #includes
+	foreach(include_file ${include_files})
+		get_filename_component(file_name ${include_file} NAME)
+		get_filename_component(file_name_we ${include_file} NAME_WE)
+		string(TOUPPER ${file_name_we} INC_FILE_NAME)
+		set(include_guard "${PROJECT_WRAPPER_MAIN_INC_GUARD}_${INC_FILE_NAME}_HPP")
+		set(include_statements "${PROJECT_WRAPPER_MAIN_INC_RELPATH}/${file_name}\"\n")
+		configure_file(${PROJECT_WRAPPER_TEMPLATE_FILE} "${PROJECT_WRAPPER_MAIN_INC_DIR}/${file_name}")
+		unset(INC_FILE_NAME)
```

Comment:
I don't think you need these `unset` statements, these are local variables which you're also setting immediately before they're used.

Reply:
> I don't think you need these `unset` statements, these are local variables which you're also setting immediately before they're used.

Updated latest version with all review comments, kindly check thanks!!